crates/state/src/values/color.rs

```diff
@@ -72,127 +85,129 @@ impl DisplayColor for Color {
     }
 }
 
-fn parse_rgb(color: &str) -> Result<Color, ParseError> {
-    if !color.ends_with(')') {
-        return Err(ParseError);
-    }
+fn parse_number_as<T: TryFrom<i64>>(token: &Token) -> Option<T> {
+    token.try_as_i64().and_then(|value| T::try_from(value).ok())
+}
 
-    let color = color.replacen("rgb(", "", 1).replacen(')', "", 1);
-
-    let mut colors = color.split(',');
-
-    let r = colors
-        .next()
-        .ok_or(ParseError)?
-        .trim()
-        .parse::<u8>()
-        .map_err(|_| ParseError)?;
-    let g = colors
-        .next()
-        .ok_or(ParseError)?
-        .trim()
-        .parse::<u8>()
-        .map_err(|_| ParseError)?;
-    let b = colors
-        .next()
-        .ok_or(ParseError)?
-        .trim()
-        .parse::<u8>()
-        .map_err(|_| ParseError)?;
-    let a: Option<&str> = colors.next();
-
-    // There should not be more than 4 components.
-    if colors.next().is_some() {
-        return Err(ParseError);
-    }
+fn parse_rgb(parser: &mut Parser) -> Result<Color, ParseError> {
+    parser.consume(&Token::ParenOpen)?;
+
+    let red = parser.consume_map(parse_number_as)?;
+
+    parser.consume(&Token::Comma)?;
+
+    let green = parser.consume_map(parse_number_as)?;
+
+    parser.consume(&Token::Comma)?;
+
+    let blue = parser.consume_map(parse_number_as)?;
+
+    let color = if parser.try_consume(&Token::Comma) {
+        let alpha = parser.consume_if(Token::is_i64_or_f32).and_then(|token| {
+            if token.is_i64() {
+                u8::try_from(token.into_i64()).map_err(|_| ParseError)
+            } else {
+                Ok((token.into_f32() * 255.0).round().clamp(0.0, 255.0) as u8)
```

Comment:
you could add a `into_u8` instead of making the conversion ere

Reply:
Instead of `into_u8', I think `try_as_u8' would be more appropriate.

Reply:
Ah yes 💯